portfolio/src/main/java/com/google/sps/servlets/AddComment.java

```diff
@@ -0,0 +1,41 @@
+package com.google.sps.servlets;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import java.io.IOException;
+import java.util.ArrayList;
+import com.google.gson.Gson;
+import com.google.sps.data.Comment;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+
+@WebServlet("/add-comment")
+public class AddComment extends HttpServlet {
+
+   @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+
+    // Take the input from the form.
+    String username = request.getParameter("username");
+    String message= request.getParameter("message");
+    long timestamp = System.currentTimeMillis();
+
+    // Create a comment
+    Entity commentEntity = new Entity("Comment");
```

Comment:
Where is Entity class defined? Can we encapsulate the creation in its constructor?

Reply:
The entity class is used to define user defined classes(here Comment Class). It is defined in the data folder's Comment file. And because it is a user defined class its creation cannot be encapsulated in its constructor.